games/devtest/mods/unittests/async_env.lua

```diff
@@ -207,3 +207,39 @@ local function test_vector_preserve(cb)
 	end, {vec})
 end
 unittests.register("test_async_vector", test_vector_preserve, {async=true})
+
+local function test_async_job_replacement(cb)
+	local capacity = core.get_async_threading_capacity()
+	local running = capacity
+	for _ = 1, capacity do
+		core.handle_async(function()
+			local t = core.get_us_time()
+			core.ipc_poll("unittests:end_blocking", 100)
+		end, function()
+			running = running - 1
+			if running <= 0 then
+				core.ipc_set("unittests:end_blocking", nil)
+			end
```

Comment:
what's the point of this?

Reply:
I added this mainly in case someone wants to run the unittest for a second time. Otherwise `unittests:end_blocking` would then be set to true (by the first `/unittests` run) when the blocking jobs are run for a second time; this would immediately un-block the async threads which can make the unittests fail.

Reply:
this would make more sense at the very end of the test imo

Reply:
Like this? https://github.com/minetest/minetest/pull/14602/commits/9ba191cbc03629a80b1a44102ff08620b680b735